pkg/scheduler/plugin.go

```diff
@@ -615,13 +615,12 @@ func (k *deleter) deleteOne(pod *Pod) error {
 	}
 }
 
-// Create creates a scheduler plugin and all supporting background functions.
+// Create a scheduler plugin and all supporting background functions.
 func (k *KubernetesScheduler) NewPluginConfig(terminate <-chan struct{}, mux *http.ServeMux) *PluginConfig {
-
 	// Watch and queue pods that need scheduling.
 	updates := make(chan queue.Entry, k.schedcfg.UpdatesBacklog)
 	podUpdates := &podStoreAdapter{queue.NewHistorical(updates)}
-	reflector := cache.NewReflector(createAllPodsLW(k.client), &api.Pod{}, podUpdates, 0)
+	reflector := cache.NewReflector(k.podsListWatch, &api.Pod{}, podUpdates, 0)
```

Comment:
instead of adding `podsListWatch` as a scheduler field, could we instead add an optional 3rd parameter to NewPluginConfig, `makePodsWatcher func() *cache.ListWatch`? Then, in NewPluginConfig:
```go
if makePodsWatcher == nil {
  makePodsWatcher = func() *cache.ListWatch { return createAllPodsLW(k.client) }
}
reflector := cache.NewReflector(makePodsWatcher(), &api.Pod{}, podUpdates, 0)
```
this allows us to:
- keep createAllPodsLW private
- avoids leaking podListWatch to the KubernetesScheduler struct
- keep setting up awesome unit/integration tests